libraries/AP_HAL/CANIface.h

```diff
@@ -191,10 +193,7 @@ class AP_HAL::CANIface
     }
 
     //Get status info of the interface
-    virtual uint32_t get_stats(char* data, uint32_t max_size)
-    {
-        return 0;
-    }
+    virtual void get_stats(ExpandingString *str) {}
```

Comment:
Being void means you can't distinguish between, "I failed to get the data" and "there are no entries"

Reply:
i've fixed this in caller by adding has_failed_allocation(). That avoids each method having to care